src/main/java/org/jetbrains/research/intellijdeodorant/ide/ui/GodClassUserInputDialog.java

```diff
@@ -0,0 +1,182 @@
+        this.javaLangClassNames.add("Number");
+        this.javaLangClassNames.add("Object");
+        this.javaLangClassNames.add("Package");
+        this.javaLangClassNames.add("Process");
+        this.javaLangClassNames.add("Runtime");
+        this.javaLangClassNames.add("Short");
+        this.javaLangClassNames.add("String");
+        this.javaLangClassNames.add("StringBuffer");
+        this.javaLangClassNames.add("StringBuilder");
+        this.javaLangClassNames.add("System");
+        this.javaLangClassNames.add("Thread");
+        this.javaLangClassNames.add("Void");
+
+        String packageName = PsiUtil.getPackageName(refactoring.getSourceClass());
+        parentPackage = JavaPsiFacade.getInstance(refactoring.getProject()).findPackage(packageName);
+
+        initialiseControls();
+        setTitle(TITLE);
+        init();
+    }
+
+    @Nullable
+    @Override
+    protected JComponent createNorthPanel() {
+        placeControlsOnPanel();
+        return mainPanel;
+    }
+
+    @Nullable
+    @Override
+    protected JComponent createCenterPanel() {
+        return null;
+    }
+
+    private void initialiseControls() {
+        extractedClassNameField.setText(refactoring.getExtractedTypeName());
+        extractedClassNameField.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                handleInputChanged(extractedClassNameField);
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                handleInputChanged(extractedClassNameField);
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                handleInputChanged(extractedClassNameField);
+            }
+        });
+
+        restoreButton.setText(RESTORE_DEFAULT);
+
+        restoreButton.addActionListener(e -> {
+            extractedClassNameField.setText(refactoring.getDefaultExtractedTypeName());
+        });
+
+        handleInputChanged(extractedClassNameField);
+    }
+
+    private void placeControlsOnPanel() {
+        FormBuilder builder = FormBuilder.createFormBuilder()
+                .addComponent(
+                        JBLabelDecorator.createJBLabelDecorator(RefactorJBundle.message("extract.class.from.label", PsiTreeUtil.findChildOfType(refactoring.getSourceFile(), PsiClass.class)))
+                                .setBold(true))
+                .addLabeledComponent(RefactorJBundle.message("name.for.new.class.label"), extractedClassNameField, UIUtil.LARGE_VGAP);
+
+        JComponent emptyComponent = new JComponent() {
+            @Override
+            public void setInheritsPopupMenu(boolean value) {
+                super.setInheritsPopupMenu(value);
+            }
+        };
+
+        builder.addLabeledComponent(restoreButton, emptyComponent);
+
+        mainPanel = builder.addVerticalGap(5).getPanel();
```

Comment:
What is a magic number?

Reply:
Magic straight from the IDEA source (everywhere I looked they just write `5`). Well, ok, extracted static final field.